src/select/Select.ts

```diff
@@ -0,0 +1,155 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v, w } from '@dojo/widget-core/d';
+import Label, { LabelOptions } from '../label/Label';
+import * as css from './styles/select.css';
+
+/**
+ * @type SelectProperties
+ *
+ * Properties that can be set on a Select component
+ *
+ * @property describedBy	ID of an element that provides more descriptive text
+ * @property disabled			Prevents the user from interacting with the form field
+ * @property formId				ID of a form element associated with the form field
+ * @property invalid			Indicates the value entered in the form field is invalid
+ * @property label				Label settings for form label text, position, and visibility
+ * @property name					The form widget's name
+ * @property options	object of select options in the format [key: value]: option name
+ * @property readOnly			Allows or prevents user interaction
+ * @property required			Whether or not a value is required
+ * @property value				The current value
+ * @property onBlur				Called when the input loses focus
+ * @property onChange			Called when the node's 'change' event is fired
+ * @property onClick			Called when the input is clicked
+ * @property onFocus			Called when the input is focused
+ * @property onKeyDown		Called on the input's keydown event
+ * @property onKeyPress		Called on the input's keypress event
+ * @property onKeyUp			Called on the input's keyup event
+ * @property onMouseDown	Called on the input's mousedown event
+ * @property onMouseUp		Called on the input's mouseup event
+ * @property onTouchStart	Called on the input's touchstart event
+ * @property onTouchEnd		Called on the input's touchend event
+ * @property onTouchCancel	Called on the input's touchcancel event
+ */
+export interface SelectProperties extends ThemeableProperties {
```

Comment:
I'm still not entirely convinced. We're already cutting down on browser functionality by not supporting `<opt-group>`, and I'm fine being a little opinionated about good and bad widget patterns, as lot as it's possible to achieve the functionality users might want. If we add a separate multi-select component with good usability, what's the added benefit of supporting multi-select here with poor usability? Additionally, supporting multi-select here should probably start a discussion on whether we want to support non-native styling. I added some dom structure to allow better custom styles, but it may make sense to remove it if we're also doing multi-select.